manticore/core/manticore.py

```diff
@@ -278,6 +283,69 @@ def _join_workers(self):
             while len(self._workers) > 0:
                 self._workers.pop().join()
 
+    #
+    # Symbol resolution
+    #
+
+    def _addr_by_symline(dwarfinfo, symbols, line):
+        '''
+        Return address for a symbol and line it is on through DWARF
+        debug info
+        '''
+        for symbol in symbols:
+            for CU in dwarfinfo.iter_CUs():
+                lineprog = dwarfinfo.line_program_for_CU(CU)
+                prevstate = None
+
+                for entry in lineprog.get_entries():
+                    # check for new state assignment
+                    if entry.state is None or entry.state.end_sequence:
+                        continue
+                    # check between address range between prev and current state containing address
+                    if prevstate and prevstate.address <= symbol['st_value'] < entry.state.address:
+                        state_line = prevstate.line
+                        if int(state_line) is line:
+                            return symbol['st_value']
+                    prevstate = entry.state
+        
+        return None, None
+
+
+    def resolve(self, symbol, line=None):
+        '''
+        A helper method used to resolve a symbol name into a memory address when
+        injecting hooks for analysis. 
+
+        :param symbol: function name to be resolved
+        :type symbol: string
+
+        :param line: if more functions present, optional line number can be included
+        :type line: int or None
+        '''
+        with open(self._path, 'rb') as f:
+
+            elffile = ELFFile(f)
+            if elffile is None:
+                return NotImplementedError("Symbols aren't supported")
+ 
+            if not elffile.has_dwarf_info():
+                raise NotImplementedError("DWARF debug info not supported")
```

Comment:
The error message is a bit misleading, maybe sth like this?:
```suggestion
                raise NotImplementedError(f"The {self._path} ELF file is missing DWARF debug info")
```